matplotlib2tikz/legend.py

```diff
@@ -181,6 +212,30 @@ def draw_legend(data, obj):
             alignment = None
             break
 
+    # Set color of lines in legend
+    set_colors = False
+    temp = []
+    for i in range(obj.legendHandles.__len__()):
+        try:
+            data, legend_color, _ = mycol.mpl_color2xcolor(data, \
+                                                           obj.legendHandles[
+                                                               i].get_color())
+            temp.append('\\addlegendimage{no markers, %s}' \
+                                         % legend_color + '\n')
+
+        except:
+            set_colors = False
+            break
+        if obj.legendHandles[i].get_color() != 'k':
```

Comment:
Also, this overrides `set_colors` in every iterations. Probably not what you want.